### PR TITLE
Fixed issue #257. 

### PR DESCRIPTION
**Pull request by abhishekkumar2718**
Returns None if user.get_display_name response has error code 404. Really trivial stuff 
#257

### CODE REVIEW

**Comment by non-Jedi** [comment]
Could you signoff on this per CONTRIBUTING.rst please? Then I'll merge as this is strictly an improvement to current behavior.

However, in general this isn't how we should be handling displaynames (which can change and so should only be cached by `sync` methods). Especially considering that according to current design of code a `User` object can be associated with a particular room which may have a different displayname than the user's "main" displayname. None of this is for you but is intended as a note to self for me.

File: matrix_client/user.py [hunk 1]
```diff
@@ -33,7 +34,11 @@ def get_display_name(self):
             str: Display Name
         """
         if not self.displayname:
-            self.displayname = self.api.get_display_name(self.user_id)
+            try:
+                self.displayname = self.api.get_display_name(self.user_id)
+            except MatrixRequestError as e:
+                if e.code == 404:
```
Comment (Zil0) [inline matrix_client/user.py]: This has a behavior that may not be intended. If the request fails with any other error code, the method still returns `None`. Currently it is equivalent to
```
except MatrixRequestError:
    pass
```
Maybe you wanted to do:
```
except MatrixRequestError as e:
    if e.code != 404:
        raise
```

**Comment by abhishekkumar2718** [comment]
Signed-Off-By: Abhishek Kumar abhishekkumar2718@gmail.com

**Comment by non-Jedi** [comment]
Thanks for taking the time to do this. This PR is actually obsoleted by #259 which makes this whole thing better. I really appreciate you taking the time to contribute and hope you'll continue to do so in the future. :)

**Pull request closed without merging**